src/benchmarklib/tpch/tpch_query_generator.cpp

```diff
@@ -0,0 +1,33 @@
+#include "tpch_query_generator.hpp"
+
+#include <numeric>
+
+#include "tpch_queries.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+TPCHQueryGenerator::TPCHQueryGenerator() {
+  _generate_names();
+  _selected_queries.resize(22);
+  std::iota(_selected_queries.begin(), _selected_queries.end(), QueryID{0});
+}
+
+TPCHQueryGenerator::TPCHQueryGenerator(const std::vector<QueryID>& selected_queries) {
+  _generate_names();
+  _selected_queries = selected_queries;
+}
+
+void TPCHQueryGenerator::_generate_names() {
+  _query_names.reserve(22);
+  for (auto i = 0; i < 22; ++i) {
+    _query_names.emplace_back(std::string("TPC-H ") + std::to_string(i + 1));
+  }
+}
+
+std::string TPCHQueryGenerator::build_query(const QueryID query_id) {
```

Comment:
Just for my understanding: Once we support actual random query generation for TPC-H, this is the function that would return a randomly generated variant of the specified query?

Reply:
Yes. Or, more specifically, an `EXECUTE` statement with the proper parameters.

Reply:
And how would the "PREPARE" statements be generated?